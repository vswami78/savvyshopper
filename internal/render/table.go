package render

import (
	"fmt"
	"io"
	"text/tabwriter"

	"savvyshopper/domain"
)

// Table writes the offers to w in a tabular format.
func Table(w io.Writer, offers []domain.Offer) error {
	tw := tabwriter.NewWriter(w, 0, 0, 2, ' ', 0)
	fmt.Fprintln(tw, "Title\tPrice\tRetailer\tURL")
	for _, offer := range offers {
		title := offer.Title
		if len(title) > 60 {
			title = title[:60]
		}
		fmt.Fprintf(tw, "%s\t$%.2f\t%s\t%s\n", title, offer.Price, offer.Retailer, offer.URL)
	}
	return tw.Flush()
}
