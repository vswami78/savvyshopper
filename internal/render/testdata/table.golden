Title                                                         Price   Retailer  URL
Short Title                                                   $10.99  Amazon    https://example.com/1
Very Long Title That Should Be Truncated Because It Exceeds   $20.99  Walmart   https://example.com/2 